pkg/apis/syn/v1alpha1/tenant_types.go

```diff
@@ -122,3 +125,121 @@ func (t *Tenant) ApplyTemplate(template *TenantTemplate) error {
+// The ServiceAccount has its meta data set in a way, that it is clear this ServiceAccount is related to the tenant.
+func (t *Tenant) NewServiceAccount() *corev1.ServiceAccount {
+	sa	:= &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      t.Name,
+			Namespace: t.Namespace,
+		},
+	}
+	setManagedByLabel(sa)
+	setOwnerReference(sa, t)
+
+	return sa
+}
+
+// NewRole returns a new Role object.
+// The Role controls access to the tenant and its related clusters.
+func (t *Tenant) NewRole() *rbacv1.Role {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      t.Name,
+			Namespace: t.Namespace,
+		},
+	}
+	setManagedByLabel(role)
+	setOwnerReference(role, t)
+	roleEnsureRules(role)
+
+	return role
+}
+
+// SynchronizeRole updates Role so it meets the needs as an object to grant access to the tenant.
+// The function will return true, if the role got altered.
+func (t *Tenant) SynchronizeRole(role *rbacv1.Role) bool {
+	return roleSynchronizeResourceNames(role, t.Name)
+}
+
+// NewRoleBinding returns a new RoleBinding object.
+// Its intention is to link subjects to the role created with `Tenant.NewRole()`.
+func (t *Tenant) NewRoleBinding() *rbacv1.RoleBinding {
+	builder := v1.NewRoleBinding(t.Name, t.Namespace)
+	builder.SAs(t.Namespace, t.Name)
+
+	binding := builder.BindingOrDie()
+	setManagedByLabel(&binding)
+	setOwnerReference(&binding, t)
+
+	return &binding
+}
+
+func roleEnsureRules(role *rbacv1.Role) int {
+	if role.Rules == nil {
+		role.Rules = []rbacv1.PolicyRule{}
+	}
+	i, err := roleGetRuleIndex(role)
+	if err != nil {
+		rule := rbacv1.PolicyRule{
+			APIGroups: []string{SchemeGroupVersion.Group},
+			Verbs:     []string{"get"},
+			Resources: []string{"tenants", "clusters"},
+		}
+		role.Rules = append(role.Rules, rule)
+		i = len(role.Rules) - 1
+	}
+
+	return i
+}
+
+func roleSynchronizeResourceNames(role *rbacv1.Role, name string) bool {
+	i := roleEnsureRules(role)
+
+	if contains(role.Rules[i].ResourceNames, name) {
+		return false
+	}
+
+	role.Rules[i].ResourceNames = append(role.Rules[i].ResourceNames, name)
+
+	return true
+}
+
+func roleGetRuleIndex(role *rbacv1.Role) (int, error) {
+	for i, rule := range role.Rules {
+		matchGroups := reflect.DeepEqual(rule.APIGroups, []string{SchemeGroupVersion.Group})
+		matchVerbs := reflect.DeepEqual(rule.Verbs, []string{"get"})
+		matchResources := contains(rule.Resources, "tenants") && contains(rule.Resources, "clusters")
+		if  matchGroups && matchVerbs && matchResources {
+			return i, nil
+		}
+	}
+	return 0, fmt.Errorf("no matching rule")
+}
+
+func setManagedByLabel(obj metav1.Object) {
+	obj.SetLabels(map[string]string{
+		"app.kubernetes.io/managed-by": "lieutenant",
+	})
+}
+
+func setOwnerReference(obj metav1.Object, owner *Tenant) {
+	obj.SetOwnerReferences([]metav1.OwnerReference{
+		{
+			APIVersion: SchemeGroupVersion.Identifier(),
+			Kind: owner.Kind,
+			Name: owner.Name,
+		},
+	})
```

Comment:
owner.Kind is actually empty if the object was created with "New...".
But there's a utility function
```suggestion
	controllerutil.SetOwnerReference(owner, obj, scheme)
```
We may need more and more to have a common scheme variable somewhere, like https://github.com/vshn/k8up/blob/master/main.go#L33